rules-reviewed/camel3/camel2/xml-generic-information.windup.xml

```diff
@@ -30,7 +30,103 @@
+                    </xmlfile>
+                </or>
+            </when>
+            <perform>
+                <hint title="`custom` load balancer was renamed `customLoadBalancer`" effort="1"
+                      category-id="mandatory">
+                    <message>`custom` load balancer was renamed `customLoadBalancer`
+
+                    </message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_xml_dsl_migration"
+                          title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="camel.customLoadBalancerReplacementSpring">
+                        <replacement>c:customLoadBalancer</replacement>
+                        <search>c:custom</search>
+                    </quickfix>
+                    <quickfix type="REPLACE" name="camel.customLoadBalancerReplacementBluePrint">
+                        <replacement>b:customLoadBalancer</replacement>
+                        <search>b:custom</search>
+                    </quickfix>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="xml-generic-information-00003">
+            <when>
+                <or>
+                    <xmlfile matches="//*/b:marshal/b:zipFile">
+                        <namespace prefix="b" uri="http://camel.apache.org/schema/blueprint"/>
+                    </xmlfile>
+                    <xmlfile matches="//*/b:unmarshal/b:zipFile">
+                        <namespace prefix="b" uri="http://camel.apache.org/schema/blueprint"/>
+                    </xmlfile>
+                    <xmlfile matches="//*/c:marshal/c:zipFile">
+                        <namespace prefix="c" uri="http://camel.apache.org/schema/spring"/>
+                    </xmlfile>
+                    <xmlfile matches="//*/c:unmarshal/c:zipFile">
+                        <namespace prefix="c" uri="http://camel.apache.org/schema/spring"/>
+                    </xmlfile>
+                </or>
+            </when>
+            <perform>
+                <hint title="`zipFile` data format was renamed to `zipfile`" effort="1"
+                      category-id="mandatory">
+                    <message>`zipFile` data format was renamed to `zipfile`
+
+                    </message>
+                    <link href="https://camel.apache.org/manual/latest/camel-3-migration-guide.html#_xml_dsl_migration"
+                          title="Camel 3 - Migration Guide"/>
+                    <quickfix type="REPLACE" name="camel.zipfile.dataformat.replacement.spring">
+                        <replacement>c:zipfile</replacement>
+                        <search>c:zipFile</search>
+                    </quickfix>
+                    <quickfix type="REPLACE" name="camel.dataformat.zipfile.replacement.blueprint">
+                        <replacement>b:zipfile</replacement>
+                        <search>b:zipFile</search>
+                    </quickfix>
```

Comment:
As `quickfix` does text searches I would suggest replacing the 2 quickfixes by 1 with this code :
```
                    <quickfix type="REPLACE" name="camel.dataformat.zipfile.replacement">
                        <replacement>zipfile</replacement>
                        <search>zipFile</search>
                    </quickfix>
```